zerver/lib/send_email.py

```diff
@@ -325,10 +327,16 @@ def initialize_connection(connection: Optional[BaseEmailBackend] = None) -> Base
             status = connection.connection.noop()[0]
         except Exception:
             status = -1
-        if status != 250:
+        if status != 250 or (
+            settings.SMTP_MAX_CONNECTION_MINUTES != 0
+            and hasattr(connection, "END_MAX_CONNECTION_MINUTES")
+            and connection.END_MAX_CONNECTION_MINUTES + settings.SMTP_MAX_CONNECTION_MINUTES * 60
+            < int(time.time())
+        ):
```

Comment:
I think this logic is better placed on `zerver/worker/queue_processors.py`, where we can place the new information on the queue worker, rather than jamming it into an object that doesn't have a property meant to store it.  mypy is rightly upset at you making up properties of objects to assign into. :)

Importantly, this also lets us make a key behaviour change -- as yoiu've written it now, setting `SMTP_MAX_CONNECTION_MINUTES = 0` means that we open the connection, send one email, and then _leave the connection open between email sends_, even though the next email send will close and re-open it.  We should be a good citizen and close the connection immediately if we're not going to use it.  We can do this by not using `initialize_connection` at all, in `EmailSendingWorker` -- if we don't pass a `connection=self.connection` into `send_email()` then it will make a one-off connection, send the email, and immediately close the connection.